tajo-core/src/main/java/org/apache/tajo/engine/planner/physical/SeqScanExec.java

```diff
@@ -167,23 +162,55 @@ public void init() throws IOException {
       projected = outSchema;
     }
 
-    initScanner(projected);
-    super.init();
-
-    if (plan.hasQual()) {
-      if (scanner.isProjectable()) {
-        qual.bind(context.getEvalContext(), projected);
-      } else {
-        qual.bind(context.getEvalContext(), inSchema);
-      }
+    return projected;
+  }
 
+  private void initScanIterator() {
+    // We should use FilterScanIterator only if underlying storage does not support filter push down.
```

Comment:
You seem to want to use an unified interface for both index scanner and this work. I'm going to add the storage-related method to determine sub filter expressions which can be push down into a specific storage. Index scanner will be affected by this method in the same manner.